astroid/brain/brain_typing.py

```diff
@@ -344,6 +345,29 @@ def infer_tuple_alias(
     return iter([class_def])
 
 
+def _looks_like_typing_cast(node: Call) -> bool:
+    return (
+        isinstance(node, Call)
+        and isinstance(node.func, Name)
+        and node.func.name == "cast"
```

Comment:
```suggestion
        and (
            isinstance(node.func, Name)
            and node.func.name == "cast"
            or isinstance(node.func, Attribute)
            and node.func.attrname == "cast"
        )
```
`cast` can also be called with `typing.cast` in which case it's an `Attribute`.